cmd/ctr/commands/images/pull.go

```diff
@@ -119,19 +119,20 @@ command. As part of this process, we do the following:
 			if err != nil {
 				return err
 			}
-
-			// Set unpack configuration
-			for _, platform := range p {
-				sopts = append(sopts, image.WithUnpack(platform, cliContext.String("snapshotter")))
+			allPlatforms := cliContext.Bool("all-platforms")
+			if len(p) > 0 && allPlatforms {
+				return errors.New("cannot specify both --platform and --all-platforms")
```

Comment:
This is probably out-of-scope for this PR, but are there other subcommands using `all-platforms` that would benefit from this check?

Reply:
Other commands just ignore `--platform`s if both are specified, e.g., in `Fetch`:

https://github.com/containerd/containerd/blob/e514bae935a24e78c1e66ae4d609040b9f14a8e4/cmd/ctr/commands/content/fetch.go#L137-L143

I think error early is better than ignore one flag, but am okay to keep it the same with other commands